src/Sulu/Bundle/MediaBundle/Resources/js/containers/Form/fields/MediaSelection.js

```diff
@@ -25,6 +25,10 @@ class MediaSelection extends React.Component<FieldTypeProps<Value>> {
             } = {},
         } = schemaOptions;
 
+        if (value != null && (typeof value !== 'object' || !Array.isArray(value.ids))) {
```

Comment:
Which values are you trying to exclude here? Especially with the first non-typesafe comparison?

Would it maybe be easier to read if we turn that into a set of acceptable values?

```javascript
if (!(value === null || (typeof value === 'object' && Array.isArray(value.ids)) {}
```

I am assuming here the value must be `null` or an object containing an `ids` array.

Resp. your current solution would also work for me if the first check is type strict. But if we are using just `!=` it almost always feels like there is a trap :see_no_evil:

Reply:
And isn't there a similar check missing in the `SingleMediaSelection` field type as well?

Reply:
i initially used `value !== undefined` instead of `value != null`, but then flow complains that `Property ids is missing in null or undefined`. This is because the `value` prop is of the type `?T` which can be `null` or `undefined` apparently 🙈

Reply:
flow will not complain when using `value !== null` or your suggestions, but the application will crash if the value is set to `undefined` (which it is when opening the add form). This is because undefined is not strictly equal to null and `typeof undefined` is `undefined`. 

So i am not sure how to write the condition correctly. Should I just check for `null` and for `undefined`? 😅

Reply:
Regarding to the `SingleMediaSelection` - this is not a big problem there because the `SingleMediaSelection` will not crash if the given value has no `id` property. It will just act like nothing is selected. But I can add a similar check there too if you think that would be important

Reply:
> flow will not complain when using `value !== null` or your suggestion, but the application will crash if the value is set to `undefined` (which it is when opening the add form). This is because undefined is not strictly equal to null and `typeof undefined` is `"undefined"`.
> 
> So i am not sure how to write the condition correctly. Should I just check for `null` and for `undefined`?

I think checking for both would make the code easier to understand, don't you think so too? This way you have to know how the casting in JS works, which probably nobody knows by heart :see_no_evil: Resp. we could maybe also adjust the type passing that in? Can it be `null` or undefined? If it can't be `null`, we could also change the type from `?T` to `T | undefined`, right?

> Regarding to the `SingleMediaSelection` - this is not a big problem there because the `SingleMediaSelection` will not crash if the given value has no `id` property. It will just act like nothing is selected. But I can add a similar check there too if you think that would be important

Well, if it is working correctly there I don't see any point in doing so.

Reply:
> I think checking for both would make the code easier to understand, don't you think so too? This way you have to know how the casting in JS works, which probably nobody knows by heart 🙈

Thats okay for me 🙂 

> Resp. we could maybe also adjust the type passing that in? Can it be null or undefined? If it can't be null, we could also change the type from ?T to T | undefined, right?

I am a bit hesitant about changing this, because it is part of the `FieldTypeProps` type and therefore would be changed for each of our field types 🙈